diff -ur gpgme-0.3.2/tests/gpg/mkdemodirs gpgme-0.3.2-MacOSX/tests/gpg/mkdemodirs
--- gpgme-0.3.2/tests/gpg/mkdemodirs	Tue Oct 16 23:08:55 2001
+++ gpgme-0.3.2-MacOSX/tests/gpg/mkdemodirs	Tue Feb 12 21:33:25 2002
@@ -25,7 +25,7 @@
 echo -n "Creating:"
 for name in $NAMES; do
     echo -n " $name"
-    [ -d $name ] && rm -r $name
+    [ -d $name ] && rm -r $name || true
     mkdir $name
     $GPGDEMO --export-secret-key -o - $name > $name/Secret.gpg
     $GPG --homedir $name --allow-secret-key-import --import $name/Secret.gpg
